netplan_cli/cli/utils.py

```diff
@@ -196,10 +196,43 @@ def get_interface_driver_name(interface, only_down=False):  # pragma: nocover (c
     return driver_name
 
 
-def get_interface_macaddress(interface):
-    # return an empty list (and string) if no LL data can be found
-    link = netifaces.ifaddresses(interface).get(netifaces.AF_LINK, [{}])[0]
-    return link.get('addr', '')
+def _get_permanent_macaddress(interface: str) -> str:
+    mac = ""
+    try:
+        out = subprocess.check_output(['ethtool', '-P', interface]).decode('utf-8')
+        split = out.split(': ')
+        if len(split) == 2 and split[1].strip() != 'not set':
+            mac = split[1].strip()
+    except Exception:
+        return mac
+
+    return mac
+
+
+def _get_macaddress(interface: str) -> str:
+    try:
+        with open(f'/sys/class/net/{interface}/address') as f:
+            return f.read().strip()
+    except Exception:
+        return ""
```

Comment:
question (non-blocking): Is it essential to return the empty string here? If so, why? Otherwise, I'd prefer returning `None` in the error case, as I think that makes the intention a bit more clear.

Reply:
The only reason is because I annotated the function with `-> str`. I've been trying to avoid this typing confusion in Python (None is not a str). I will adopt `Optional[str]` here for the type hint and return None.